src/charm.py

```diff
@@ -219,6 +265,37 @@ def _generate_kiali_layer() -> Layer:
                 },
             }
         )
+        LOGGER.debug(f"Kiali layer: {layer}")
+        return layer
+
+    def _get_grafana_urls(self) -> GrafanaUrls:
+        """Return the urls for the related Grafana.
+
+        This retrieves the grafana urls from the grafana-metadata relation, returning:
+        * internal_url: GrafanaMetadataAppData.direct_url
+        * external_url: GrafanaMetadataAppData.ingress_url
+
+        If GrafanaMetadataAppData.ingress_url is not available, it will default to GrafanaMetadataAppData.direct_url.
+
+        Raises:
+          ConfigurationBlockingError: If no grafana is related to this application
+          ConfigurationWaitingError: If a grafana is related to this application, but its data is incomplete.
+        """
+        LOGGER.debug("Getting Grafana configuration")
+        if len(self._grafana_metadata.relations) == 0:
+            raise BlockedStatusError("No grafana available over the grafana-metadata relation")
```

Comment:
Should this really cause us to block? Why is this relation required? As far as I am aware Kiali works just fine with only Prometheus.

Reply:
Refactored to make it clearer this is not going to block the charm